pallets/crowdloan-claim/src/lib.rs

```diff
@@ -0,0 +1,850 @@
+    PalletId,
+};
+use frame_system::ensure_root;
+use sp_core::crypto::AccountId32;
+use sp_core::storage::ChildInfo;
+use sp_core::Hasher;
+use sp_runtime::{
+    sp_std::{vec, vec::Vec},
+    traits::{AccountIdConversion, Verify, Zero},
+    MultiSignature,
+};
+use sp_state_machine::read_child_proof_check;
+use sp_std::convert::TryInto;
+use sp_trie::StorageProof;
+
+// Re-export in crate namespace (for runtime construction)
+pub use pallet::*;
+
+// Extrinsics weight information
+pub use crate::traits::WeightInfo as PalletWeightInfo;
+use frame_support::dispatch::DispatchError;
+
+// Mock runtime and unit test cases
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+// Extrinsics weight information (computed through runtime benchmarking)
+pub mod weights;
+
+// ----------------------------------------------------------------------------
+// Traits and types declaration
+// ----------------------------------------------------------------------------
+
+pub mod traits {
+    use super::*;
+
+    /// Weight information for pallet extrinsics
+    ///
+    /// Weights are calculated using runtime benchmarking features.
+    /// See [`benchmarking`] module for more information.
+    pub trait WeightInfo {
+        fn initialize() -> Weight;
+        fn claim_reward() -> Weight;
+        fn set_lease_start() -> Weight;
+        fn set_lease_period() -> Weight;
+        fn set_locked_at() -> Weight;
+        fn set_contributions_root() -> Weight;
+        fn set_crowdloan_trie_index() -> Weight;
+    }
+} // end of 'traits' module
+
+/// A type alias for the balance type from this pallet's point of view.
+type BalanceOf<T> = <T as pallet_balances::Config>::Balance;
+
+/// A type alias for crowdloan's child trie root hash, from this claim pallet's point of view.
+///
+/// When setting up the pallet via the [`initialize`] transaction, the
+/// child trie root hash containing all contributions, is transfered from
+/// the [`crowdloan`] pallet' storage to [`Contributions`] storage item
+// of this pallet.
+/// The [`Contributions`] root hash is used to check if a contributor is
+/// eligible for a reward payout and to get the amount of her/his contribution
+/// (in relay chain's native token) to a crowdloan campaign.
+type RootHashOf<T> = <T as frame_system::Config>::Hash;
+
+/// A type alias for the parachain account identifier from this claim pallet's point of view
+type ParachainAccountIdOf<T> =
+    <<T as Config>::RewardMechanism as trait_crowdloan_reward::Reward>::ParachainAccountId;
+
+/// A type alias for the contribution amount (in relay chain tokens) from this claim pallet's point of view
+type ContributionAmountOf<T> =
+    <<T as Config>::RewardMechanism as trait_crowdloan_reward::Reward>::ContributionAmount;
+
+/// Index of the crowdloan campaign inside the
+/// [crowdloan.rs](https://github.com/paritytech/polkadot/blob/77b3aa5cb3e8fa7ed063d5fbce1ae85f0af55c92/runtime/common/src/crowdloan.rs#L80)
+/// on polkadot.
+type TrieIndex = u32;
```

Comment:
No, I dont think it will be a benefit. Parity uses a u32 for their `TrieIndex` in the crowdloan and as long as this does not change we are good. In case of a change, a runtime upgrade would be necessary either way, storage migration included.